Global.h

```diff
@@ -279,442 +279,397 @@ void inline handleException()
-std::ostream & operator<<(std::ostream & out, const boost::optional<T> & opt)
+template<typename T> std::ostream & operator<<(std::ostream & out, const boost::optional<T> & opt)
 {
-	if(opt) return out << *opt;
-	else return out << "empty";
+	if(opt)
+		return out << *opt;
+	else
+		return out << "empty";
 }
 
-template<typename T>
-std::ostream & operator<<(std::ostream & out, const std::vector<T> & container)
+template<typename T> std::ostream & operator<<(std::ostream & out, const std::vector<T> & container)
 {
 	out << "[";
 	for(auto it = container.begin(); it != container.end(); ++it)
 	{
 		out << *it;
-		if(std::prev(container.end()) != it) out << ", ";
+		if(std::prev(container.end()) != it)
+			out << ", ";
 	}
 	return out << "]";
 }
 
 namespace vstd
 {
 
-	// combine hashes. Present in boost but not in std
-	template <class T>
-	inline void hash_combine(std::size_t& seed, const T& v)
-	{
-		std::hash<T> hasher;
-		seed ^= hasher(v) + 0x9e3779b9 + (seed<<6) + (seed>>2);
-	}
-
-	//returns true if container c contains item i
-	template <typename Container, typename Item>
-	bool contains(const Container & c, const Item &i)
-	{
-		return std::find(std::begin(c), std::end(c),i) != std::end(c);
-	}
+// combine hashes. Present in boost but not in std
+template<class T> inline void hash_combine(std::size_t & seed, const T & v)
+{
+	std::hash<T> hasher;
+	seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+}
 
-	//returns true if container c contains item i
-	template <typename Container, typename Pred>
-	bool contains_if(const Container & c, Pred p)
-	{
-		return std::find_if(std::begin(c), std::end(c), p) != std::end(c);
-	}
+//returns true if container c contains item i
+template<typename Container, typename Item> bool contains(const Container & c, const Item & i)
+{
+	return std::find(std::begin(c), std::end(c), i) != std::end(c);
+}
 
-	//returns true if map c contains item i
-	template <typename V, typename Item, typename Item2>
-	bool contains(const std::map<Item,V> & c, const Item2 &i)
-	{
-		return c.find(i)!=c.end();
-	}
+//returns true if container c contains item i
+template<typename Container, typename Pred> bool contains_if(const Container & c, Pred p)
+{
+	return std::find_if(std::begin(c), std::end(c), p) != std::end(c);
+}
 
-	//returns true if unordered set c contains item i
-	template <typename Item>
-	bool contains(const std::unordered_set<Item> & c, const Item &i)
-	{
-		return c.find(i)!=c.end();
-	}
+//returns true if map c contains item i
+template<typename V, typename Item, typename Item2> bool contains(const std::map<Item, V> & c, const Item2 & i)
+{
+	return c.find(i) != c.end();
+}
 
-	template <typename V, typename Item, typename Item2>
-	bool contains(const std::unordered_map<Item,V> & c, const Item2 &i)
-	{
-		return c.find(i)!=c.end();
-	}
+//returns true if unordered set c contains item i
+template<typename Item>	bool contains(const std::unordered_set<Item> & c, const Item & i)
+{

```

Comment:
This is problem with all of "nl_" options and ```nl_template_class``` in particular. Basically when newline removed it's keep that extra tab untouched even in case of colons and such. Se we'll need to manually replace some tabs with spaces once.

Though we already decide we don't want template to be on same line everywhere.